enterprise/server/metrics.py

```diff
@@ -1,43 +1,78 @@
+import logging
+import os
 from typing import Callable
 
-from prometheus_client import Gauge, make_asgi_app
+from prometheus_client import CollectorRegistry, Gauge, generate_latest
 from server.clustered_conversation_manager import ClusteredConversationManager
 
-from openhands.server.shared import (
-    conversation_manager,
-)
+import openhands.server.shared
+
+logger = logging.getLogger(__name__)
 
+# Use livesum mode for multiprocess support - this aggregates gauge values
+# across all worker processes, counting only living processes
 RUNNING_AGENT_LOOPS_GAUGE = Gauge(
     'saas_running_agent_loops',
     'Count of running agent loops, aggregate by session_id to dedupe',
     ['session_id'],
+    multiprocess_mode='livesum',
 )
 
 
 async def _update_metrics():
     """Update any prometheus metrics that are not updated during normal operation."""
-    if isinstance(conversation_manager, ClusteredConversationManager):
-        running_agent_loops = (
-            await conversation_manager.get_running_agent_loops_locally()
-        )
-        # Clear so we don't keep counting old sessions.
-        # This is theoretically a race condition but this is scraped on a regular interval.
-        RUNNING_AGENT_LOOPS_GAUGE.clear()
+    if isinstance(
+        openhands.server.shared.conversation_manager, ClusteredConversationManager
+    ):
+        running_agent_loops = await openhands.server.shared.conversation_manager.get_running_agent_loops_locally()
+        # In multiprocess mode, do NOT call clear() as it can delete memory-mapped files
+        if not os.environ.get('PROMETHEUS_MULTIPROC_DIR'):
+            RUNNING_AGENT_LOOPS_GAUGE.clear()
         # running_agent_loops shouldn't be None, but can be.
         if running_agent_loops is not None:
             for sid in running_agent_loops:
                 RUNNING_AGENT_LOOPS_GAUGE.labels(session_id=sid).set(1)
 
 
 def metrics_app() -> Callable:
```

Comment:
Actually I think we can just remove this metrics_app. I can handle that PR unless you'd like to.